cli/azd/pkg/infra/provisioning/manager.go

```diff
@@ -6,101 +6,80 @@ package provisioning
 import (
 	"context"
 	"fmt"
-	"io"
-	"log"
 
-	"github.com/azure/azure-dev/cli/azd/pkg/account"
 	"github.com/azure/azure-dev/cli/azd/pkg/alpha"
-	"github.com/azure/azure-dev/cli/azd/pkg/azureutil"
 	"github.com/azure/azure-dev/cli/azd/pkg/environment"
-	"github.com/azure/azure-dev/cli/azd/pkg/exec"
 	"github.com/azure/azure-dev/cli/azd/pkg/input"
-	"github.com/azure/azure-dev/cli/azd/pkg/spin"
-	"github.com/azure/azure-dev/cli/azd/pkg/tools"
-	"github.com/azure/azure-dev/cli/azd/pkg/tools/azcli"
+	"github.com/azure/azure-dev/cli/azd/pkg/ioc"
 )
 
 // Manages the orchestration of infrastructure provisioning
 type Manager struct {
-	azCli              azcli.AzCli
-	env                *environment.Environment
-	provider           Provider
-	writer             io.Writer
-	console            input.Console
-	accountManager     account.Manager
-	userProfileService *azcli.UserProfileService
-	subResolver        account.SubscriptionTenantResolver
-	interactive        bool
+	serviceLocator      ioc.ServiceLocator
+	env                 *environment.Environment
+	console             input.Console
+	provider            Provider
+	alphaFeatureManager *alpha.FeatureManager
+	projectPath         string
+	options             *Options
+}
+
+func (m *Manager) Initialize(ctx context.Context, projectPath string, options Options) error {
+	m.projectPath = projectPath
+	m.options = &options
+
+	provider, err := m.newProvider(ctx)
+	if err != nil {
+		return fmt.Errorf("initializing infrastructure provider: %w", err)
+	}
+
+	m.provider = provider
+	return m.provider.Initialize(ctx, projectPath, options)
 }
 
 // Prepares for an infrastructure provision operation
 func (m *Manager) Plan(ctx context.Context) (*DeploymentPlan, error) {
-	deploymentPlan, err := m.plan(ctx)
+	deploymentPlan, err := m.provider.Plan(ctx)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("planning infrastructure provisioning: %w", err)
 	}
 
 	return deploymentPlan, nil
 }
 
 // Gets the latest deployment details for the specified scope
 func (m *Manager) State(ctx context.Context) (*StateResult, error) {
-	var stateResult *StateResult
-
-	err := m.runAction(
-		ctx,
-		"Retrieving Infrastructure State",
-		m.interactive,
-		func(ctx context.Context, spinner *spin.Spinner) error {
-			queryTask := m.provider.State(ctx)
-
-			go func() {
-				for progress := range queryTask.Progress() {
-					m.updateSpinnerTitle(spinner, progress.Message)
-				}
-			}()
-
-			go m.monitorInteraction(spinner, queryTask.Interactive())
-
-			result, err := queryTask.Await()
-			if err != nil {
-				return err
-			}
-
-			stateResult = result
-
-			return nil
-		},
-	)
-
+	result, err := m.provider.State(ctx)
```

Comment:
Not sure if it matters, but I think this is a place where our UX ended up changing.  Will we continue to show a spinner for this long running operation as we had done before?

I don't think it's the end of the world if we lose this, just pointing it out.

Reply:
I will double check this one

Reply:
Fixed this.  Added back the manual usage of console messages to temporarily not cause any UX regression.  These will be updated to the vNext version of our progress messaging.